Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Customers.m

```diff
@@ -0,0 +1,269 @@
+				accessToken = authenticatedResponse.accessToken;
+			}
+			
+			error = error ?: [self extractErrorFromResponse:response json:json];
+			
+			block(accessToken, error);
+		}];
+	}
+	else {
+		block(nil, [NSError errorWithDomain:kShopifyError code:BUYShopifyError_InvalidCustomerToken userInfo:nil]);
+		return nil;
+	}
+}
+
+- (NSURLSessionDataTask *)activateCustomerWithCredentials:(BUYAccountCredentials *)credentials customerID:(NSString *)customerID customerToken:(NSString *)customerToken callback:(BUYDataCustomerTokenBlock)block
+{
+	NSURLComponents *components = [self URLComponentsForCustomerActivationWithID:customerID customerToken:customerToken];
+	NSData *data = [NSJSONSerialization dataWithJSONObject:credentials.JSONRepresentation options:0 error:nil];
+	
+	return [self putRequestForURL:components.URL body:data completionHandler:^(NSDictionary *json, NSURLResponse *response, NSError *error) {
+		if (json && error == nil) {
+			BUYAccountCredentialItem *emailItem = [BUYAccountCredentialItem itemWithKey:@"email" value:json[@"customer"][@"email"]];
+			credentials[@"email"] = emailItem;
+			[self loginCustomerWithCredentials:credentials callback:block];
+		}
+		else {
+			block(nil, nil, error);
+		}
+	}];
+}
+
+- (NSURLSessionDataTask *)resetPasswordWithCredentials:(BUYAccountCredentials *)credentials customerID:(NSString *)customerID customerToken:(NSString *)customerToken callback:(BUYDataCustomerTokenBlock)block
+{
+	NSURLComponents *components = [self URLComponentsForCustomerPasswordResetWithCustomerID:customerID customerToken:customerToken];
+	NSData *data = [NSJSONSerialization dataWithJSONObject:credentials.JSONRepresentation options:0 error:nil];
+	
+	return [self putRequestForURL:components.URL body:data completionHandler:^(NSDictionary *json, NSURLResponse *response, NSError *error) {
+		if (json && error == nil) {
+			BUYAccountCredentialItem *emailItem = [BUYAccountCredentialItem itemWithKey:@"email" value:json[@"customer"][@"email"]];
+			credentials[@"email"] = emailItem;
+			[self loginCustomerWithCredentials:credentials callback:block];
+		}
+		else {
+			block(nil, nil, error);
+		}
+	}];
+}
+
+- (NSURLSessionDataTask *)getOrdersForCustomerWithCallback:(BUYDataOrdersBlock)block
+{
+	NSURLComponents *components = [self URLComponentsForCustomerOrders];
+	return [self getRequestForURL:components.URL completionHandler:^(NSDictionary *json, NSURLResponse *response, NSError *error) {
+		NSArray *ordersJSON = json[@"orders"];
+		if (!error && ordersJSON) {
+			
+			NSMutableArray *container = [NSMutableArray new];
+			for (NSDictionary *orderJSON in ordersJSON) {
+				[container addObject:[[BUYOrder alloc] initWithDictionary:orderJSON]];
+			}
+			block([container copy], error);
+			
+		} else {
+			block(nil, error);
+		}
+	}];
+}
+
+#pragma mark - URL Formatting
+
+- (NSURLComponents *)URLComponentsForCustomers
+{
+	return [self customerURLComponents];
```

Comment:
Looks like we have two methods that do the same thing, when one is probably fine.